sqlmesh/core/model/definition.py

```diff
@@ -995,6 +1012,23 @@ class SqlModel(_SqlBasedModel):
 
     _columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None
 
+    def __getstate__(self) -> t.Dict[t.Any, t.Any]:
+        state = super().__getstate__()
+        state["__dict__"] = state["__dict__"].copy()
+        # query renderer is very expensive to serialize
+        state["__dict__"].pop("_query_renderer", None)
+        state["__dict__"].pop("column_descriptions", None)
+        private = state[PRIVATE_FIELDS]
+        private["_columns_to_types"] = None
+        return state
+
+    def copy(self, **kwargs: t.Any) -> Self:
+        model = super().copy(**kwargs)
+        model.__dict__.pop("_query_renderer", None)
```

Comment:
Ah, found it. Why not reset it using the attribute directly? Why pop it?

Reply:
because it's a cached_property and it's behavior is special